src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/InternalIPAddressCollection.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.Net.NetworkInformation
+{
+    public class InternalIPAddressCollection : IPAddressCollection
```

Comment:
As a random note: I think we will want to share this file for all platforms. I implemented something very similar separately for Linux.